pil-analyzer/src/evaluator.rs

```diff
@@ -976,8 +1005,36 @@ impl<'a, 'b, T: FieldElement, S: SymbolLookup<'a, T>> Evaluator<'a, 'b, T, S> {
             Expression::FreeInput(_, _) => Err(EvalError::Unsupported(
                 "Cannot evaluate free input.".to_string(),
             ))?,
-            Expression::StructExpression(_, _) => {
-                unimplemented!("Struct expressions are not yet supported.")
+            Expression::StructExpression(
+                _,
+                StructExpression {
+                    name: reference,
+                    fields,
+                },
+            ) => {
+                let mut exp_fields = Vec::new();
+                for NamedExpression { name, body } in fields.iter() {
+                    self.expand(body)?;
```

Comment:
Don't call expend directly on them, but put an Expand on the op stack instead.

You can expand the first item directly if you like.